pkg/mcp/server.go

```diff
@@ -885,6 +892,13 @@ func transformCloudCostSetRange(ccsr *opencost.CloudCostSetRange) *CloudCostResp
 		// Convert each cloud cost item
 		for _, item := range ccSet.CloudCosts {
 			if item == nil {
+				log.Warnf("transformCloudCostSetRange: skipping nil CloudCost item in set %s", setName)
+				continue
+			}
+
+			// Check for nil Window or nil Start/End pointers on the item
+			if item.Window.Start() == nil || item.Window.End() == nil {
+				log.Warnf("transformCloudCostSetRange: skipping CloudCost item with invalid window (nil start or end) in set %s", setName)
```

Comment:
consider printing both the start and end in all these logs to help troubleshoot, even if at least one of them will be nil, maybe the other one has a date that will clue in to the bad range